### PR TITLE
Proof request, present proof, verify proof.  Protocol implementation and full support for Indy style credentials

### LINKED ISSUE #68: replace pkg/ursa with ursa-wrapper-go
#### Helpers

- [x] NonceFromJSON
- [ ] CredentialKeyCorrectnessProofFromJSON
- [ ] BlindedCredentialSecretsFromJSON
- [ ] BlindedCredentialSecretsCorrectnessProofFromJSON
- [ ] CredentialPrivateKeyFromJSON
- [ ] (delete) CredentialPublicKeyFromJSON
- [ ] CredDefHandle

#### Issuer

`IssueCredential(issuerDID string, schemaID, credDefID, offerNonce string, blindedMasterSecret, blindedMSCorrectnessProof, requestNonce string,`
- [ ] ursa_cl_issuer_sign_credential
- [ ] ursa_cl_credential_signature_to_json
- [ ] ursa_cl_signature_correctness_proof_to_json

#### issuer_cred_def
`Finalize() error `
- [ ] ursa_cl_credential_schema_builder_add_attr
- [ ] ursa_cl_credential_schema_builder_finalize
- [ ] ursa_cl_non_credential_schema_builder_new
- [ ] ursa_cl_non_credential_schema_builder_add_attr
- [ ] ursa_cl_non_credential_schema_builder_finalize
- [ ] ursa_cl_issuer_new_credential_def
- [ ] ursa_cl_credential_public_key_to_json
- [ ] ursa_cl_credential_private_key_to_json
- [ ] ursa_cl_credential_key_correctness_proof_to_json
- [ ] ursa_cl_credential_schema_free
- [ ] ursa_cl_non_credential_schema_free
- [ ] ursa_cl_credential_private_key_free
- [ ] ursa_cl_credential_public_key_free
- [ ] ursa_cl_credential_key_correctness_proof_free

#### prover
`CreateMasterSecret(masterSecretID string) (string, error)`
- [ ] ursa_cl_prover_new_master_secret
- [ ] ursa_cl_master_secret_to_json

`CreateCredentialRequest`
- [ ] ursa_cl_prover_blind_credential_secrets
- [ ] ursa_get_current_error
- [ ] ursa_cl_blinded_credential_secrets_correctness_proof_to_json
- [ ] ursa_cl_blinded_credential_secrets_to_json
- [ ] ursa_cl_credential_secrets_blinding_factors_to_json 
- [ ] ursa_cl_blinded_credential_secrets_free
- [ ] ursa_cl_blinded_credential_secrets_correctness_proof_free
- [ ] ursa_cl_credential_secrets_blinding_factors_free

#### value builder
`Finalize() error `
- [ ] ursa_cl_credential_values_builder_add_dec_hidden
- [ ] ursa_cl_credential_values_builder_add_dec_known
- [ ] ursa_cl_credential_values_builder_add_dec_commitment
- [ ] ursa_cl_credential_values_builder_finalize

`Free() error`
- [ ] ursa_cl_credential_values_free

### LINKED ISSUE #80: Complete presentproof
Only RequestPresentation is "done" so... everything else

### PR DESCRIPTION
**Pull request by pfeairheller**
This PR contains:

- REST API exposed to request a proof presentation from an external ID
- Code to create an Indy proof and present it using the v2 present proof protocol.
- Proof verification using v2 present proof and verification of Indy proofs.
- Removal of all `cgo` calls to Ursa, replaced with calls to `ursa-wrapper-go`

Closes #68  and Closes #80

### CODE REVIEW

File: pkg/apiserver/server.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type APIServer struct {
 type provider interface {
 	KMS() kms.KeyManager
 	Store() datastore.Store
-	IndyVDR() (indy.IndyVDRClient, error)
+	IndyVDR() (indy2.VDRClient, error)
```
Comment (m00sey) [inline pkg/apiserver/server.go:43]: indy2

File: pkg/didcomm/issuer/issuer_test.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package issuer
+
+import (
+	"testing"
+)
+
+func TestIssueCredential(t *testing.T) {
+
```
Comment (m00sey) [inline pkg/didcomm/issuer/issuer_test.go:8]: This needs implementing?

**Review by m00sey — APPROVED** [review]

File: pkg/apiserver/server.go (hunk 1 above)
Comment (pfeairheller) [reply pkg/apiserver/server.go:43]: ooof

File: pkg/didcomm/issuer/issuer_test.go (hunk 2 above)
Comment (pfeairheller) [reply pkg/didcomm/issuer/issuer_test.go:269]: Implemented

**Pull request merged**